### PR TITLE
Option to switch units

### PR DESCRIPTION
**Pull request by chloe1192**
Option on the FMC to switch between KGS and LBS.

Known limitations: When loading Simbrief weight data, it will show the weights present on your OFP before accepting the data. However it will load the corrected weights for the aircraft after accepting.

### CODE REVIEW

**Comment by 747-4EVER** [comment]
Does this also change the fuel estimates on the Progress Page or are they still only in LBs?

File: html_ui/Pages/VCockpit/Instruments/Airliners/B747_8/FMC/B747_8_FMC_PerfInitPage.js [hunk 1]
```diff
@@ -2,6 +2,12 @@ class FMCPerfInitPage {
     static ShowPage1(fmc, store = {requestData: "<SEND", loadUplink: "", purgeUplink: "__FMCSEPARATOR", dataLink: "REQUEST", stepSizeLabel: "STEP SIZE", perfUplinkHeader: ""}) {
         fmc.updateFuelVars().then(() => {
             fmc.clearDisplay();
+            let units;
+            if (fmc.units == 1) {
+                units = 0
+            } else if (fmc.units == 0) {
+                units = 1;
+            }
```
[the remark this answers is not in the document]
Comment (chloe1192) [reply html_ui/Pages/VCockpit/Instruments/Airliners/B747_8/FMC/B747_8_FMC_PerfInitPage.js:10]: I'm still trying to find a way to polish this. Some sim functions have the parameter "useLbs", and some use "unitIsMetric". I think adding both to the fmc var would be cleaner, maybe?

**Comment by chloe1192** [comment]
> 
> 
> Does this also change the fuel estimates on the Progress Page or are they still only in LBs?

Progress page will be more complicated i think.

**Comment by 747-4EVER** [comment]
Yeah could be, I was just thinking about all the places that units are displayed from memory. Also there's the fuel flow gauges on the Engine EICAS page, I think they show KGs/hr by default.

**Comment by chloe1192** [comment]
> 
> 
> Yeah could be, I was just thinking about all the places that units are displayed from memory. Also there's the fuel flow gauges on the Engine EICAS page, I think they show KGs/hr by default.

I created a progress page, need to test if everything is working properly.

**Comment by zakiack** [comment]
Name: zakiack#8247
Date of testing: Current date 25/2/2021
Version of the sim: Version the tester uses
PR Tested: #140 

Testing technique: Entered weight values in kg into the FMC, using the commits as a list of places to enter weights, switched them to lbs, and verified the accuracy of them by calculating in google. Then doing the same but from lbs to kg using different numbers
Steps you've made to test the Pull Request:
1. Powered on aircraft from cold and dark
2. Entered weight values into the FMC in kgs
3. Switched them to lbs
4. Verified them
5. Entered different weight values into fmc in lbs
6. Switched them to kgs
7. Verified them

Overall Rating: Good
Conclusions of the testing: Values are perfect
Only issue is in the menu to switch the values, the heading says ATIS SRC 
Screenshot_539

**Comment by ninjomcs** [comment]
> Name: zakiack#8247
> Date of testing: Current date 25/2/2021
> Version of the sim: Version the tester uses
> PR Tested: #140
> 
> Testing technique: Entered weight values in kg into the FMC, using the commits as a list of places to enter weights, switched them to lbs, and verified the accuracy of them by calculating in google. Then doing the same but from lbs to kg using different numbers
> Steps you've made to test the Pull Request:
> 
> 1. Powered on aircraft from cold and dark
> 2. Entered weight values into the FMC in kgs
> 3. Switched them to lbs
> 4. Verified them
> 5. Entered different weight values into fmc in lbs
> 6. Switched them to kgs
> 7. Verified them
> 
> Overall Rating: Good
> Conclusions of the testing: Values are perfect
> Only issue is in the menu to switch the values, the heading says ATIS SRC
> Screenshot_539

Did you also test the progress page? Did it work fine?

**Comment by zakiack** [comment]
> Did you also test the progress page? Did it work fine?
 Yes, progress page works fine

**Comment by BobbyFuzzy** [comment]
Name: BobbyFuzzy#4615
Date of testing: 27/20/2021
Version of the sim: 1.13.16.0
PR Tested: #140
Object of test: CDU

Testing technique: Setup flight & switch between units. Convert and check figures are converted properly.
Steps you've made to test the Pull Request:
1. Setup flight from cold and dark
2. Entered perf figures, switched units
3. Checked figures were correct by conversion
4. Started flight and checked progress page figures in the same manner

Flight notes:

- Everything OK on PERF INIT.
- Progress page seems to give me minus figures after converting to lbs whilst inflight, not sure if relevant as units would unlikely be changed in flight

image

Issues (If Any): Mentioned above, may not be a concern
Overall Rating: Good
Conclusions of the testing: PR works as intended. Can switch between lbs and kgs successfully before setting up a flight.

**Comment by 747-4EVER** [comment]
Name: 747-4EVER
Date of testing: 07/03/2021
Version of the sim: 1.13.16.0
PR Tested: #140
Object of test: CDU/Displays

Testing technique: Inspect displays on ground and in-flight while switching units.

Flight notes: N/A

Issues (If Any): 
- Fuel flow gauges on Engine EICAS page do not change their units. 
- Progress page is using old codebase not in line with latest dev build, should not be difficult to merge though. 
- Noticed that the fuel remaining at destination estimate was not accurate when in KGs (Its ok in LBS). See below image:

image

ETA was approx 1.7 hours away at time of screenshot. With a fuel flow of 6TPH fuel burn to destination is 10.2T so estimate should be around 82, instead about 12 tonnes is missing.

Overall Rating: Good
Conclusions of the testing: Works mostly as intended, just needs small fixes for above issues.

**Review by 747-4EVER — CHANGES_REQUESTED** [review]
Was ready to merge, however compatibility seems to have been broken with latest sim update. The progress page no longer works and the fuel/engine EICAS page units don't change any more.

**Pull request merged**